wallet/src/de/schildbach/wallet/ui/dashpay/NotificationsActivity.kt

```diff
@@ -196,6 +206,22 @@ class NotificationsActivity : LockScreenActivity(), NotificationsAdapter.OnItemC
         lastSeenNotificationTime = lastNotificationTime
     }
 
+    private fun showHideAlert() {
+        if (notificationsAdapter.results.isNotEmpty()) {
+            val firstItem = notificationsAdapter.results[0]
+            if (!isBlockchainSynced && userAlertItem != null && firstItem.notificationItem is NotificationItemUserAlert) {
+                val newResults = notificationsAdapter.results.toMutableList()
+                newResults.remove(notificationsAdapter.results[0])
+                notificationsAdapter.results = newResults
+            } else if (isBlockchainSynced && userAlertItem != null && firstItem.notificationItem !is NotificationItemUserAlert) {
+                val newResults = arrayListOf<NotificationsAdapter.NotificationViewItem>()
+                newResults.add(NotificationsAdapter.NotificationViewItem(userAlertItem!!))
+                newResults.addAll(notificationsAdapter.results)
+                notificationsAdapter.results = newResults
+            }
+        }
+    }
```

Comment:
This feels like a hack, but there are two live data's that have to be merged to display the notifications properly.

Reply:
`MediatorLiveData` can be used for merging two live data's, although I'm not entirely sure which ones you're talking about